sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -266,47 +267,75 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
       relation,
       projects,
       filterPredicates,
-      (requestedColumns, pushedFilters) => {
-        scanBuilder(requestedColumns, selectFilters(pushedFilters).toArray)
+      (requestedColumns, _, pushedFilters) => {
+        scanBuilder(requestedColumns, pushedFilters.toArray)
       })
   }
 
-  // Based on Catalyst expressions.
+  // Based on Catalyst expressions. The `scanBuilder` function accepts three arguments:
+  //
+  //  1. A `Seq[Attribute]`, containing all required column attributes, used to handle traits like
+  //     `PrunedFilteredScan`.
+  //  2. A `Seq[Expression]`, containing all gathered Catalyst filter expressions, used by
+  //     `CatalystScan`.
+  //  3. A `Seq[Filter]`, containing all data source `Filter`s that are converted from (possibly a
+  //     subset of) Catalyst filter expressions and can be handled by `relation`.
```

Comment:
So, `Seq[Expression]` is used for data source that understand catalyst expressions. `Seq[Filter]` is used for data sources that only understand `Filter` API? If so, can we make it clear that `2` and `3` will not not used together?